LiteCore/Support/ChannelManifest.hh

```diff
@@ -0,0 +1,111 @@
+//
+// ChannelManifest.hh
+//
+// Copyright (c) 2020 Couchbase, Inc All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#pragma once
+
+#include "Channel.hh"
+
+#if ACTORS_USE_MANIFESTS
+
+#include <list>
+#include <iostream>
+#include <mutex>
+
+#ifdef ACTORS_USE_GCD
+#include <dispatch/dispatch.h>
+#endif
+
+namespace litecore::actor {
+    class Actor;
+    
+    /** A simple class to keep track of nested mailbox calls, similar to the way that Apple tracks
+     *  through GCD enqueue calls.  The way it works is as follows, and is common between both
+     *  GCD and threaded mailbox:
+     *
+     *  1. On the initial call to enqueue, or enqueueAfter, a thread local manifest is checked for existence.
+     *  =====
+     *  2. The manifest does not exist, and it is created and captured by the block that will run inside the mailbox
```

Comment:
thread local refers to [thread local storage](https://en.cppreference.com/w/cpp/language/storage_duration), which was introduced as a first class feature of C++11.  Basically, each thread sees its own version of the variable marked as such.

[captured ](https://en.cppreference.com/w/cpp/language/lambda) means that the variable is stored for a future execution.

Pseudocode:

```
int f = 3;
my_capture = () => {
    printf("%d\n", f); // f was captured
};
```

The block will only ever have a reference to one manifest (the one involved in the current "stack") and it will record its execution into that manifest.

Reply:
I am pretty familiar with thread local storage.  I think, then, that the answer to the question is "yes": each actor will have "captured" a pointer to multiple versions of the manifest.  Where are those version coalesced?

Reply:
The "actor" captures nothing.  The execution block captures the manifest.  This is an important distinction because what this is tracking is blocks of work, not actors.  In fact each manifest likely passes through multiple actors.  I don't find any coalescing needed since for any given lambda, only one manifest is visible.

Reply:
Ok.  I'll stop asking questions...

Reply:
Perhaps I just didn't understand the meaning of coalesce?  Are you suspecting a bug will happen without coalescence?